src/NLog.StructuredLogging.Json/LoggerExtensions.cs

```diff
@@ -119,10 +120,28 @@ private static void TransferDataToLogEventProperties(LogEventInfo log, object lo
             }
         }
 
+        private static void TransferContextDataToLogEventProperties(LogEventInfo log)
+        {
+            foreach (var contextItemName in MappedDiagnosticsLogicalContext.GetNames())
+            {
+                var value = MappedDiagnosticsLogicalContext.Get(contextItemName);
+                var key = contextItemName;
+                if (log.Properties.ContainsKey(contextItemName))
+                {
+                    key = "log_context_" + contextItemName;
+                }
+
+                if (!log.Properties.ContainsKey(key))
+                {
+                    log.Properties.Add(key, value);
+                }
+            }
+        }
+
         private static bool IsDictionary(object logProperties)
         {
-            var type = logProperties.GetType();
-            return type.GetTypeInfo().IsGenericType && type.GetGenericTypeDefinition() == typeof(Dictionary<,>);
+            var propsType = logProperties.GetType().GetTypeInfo();
+            return typeof(IDictionary).GetTypeInfo().IsAssignableFrom(propsType);
```

Comment:
Could the value of `typeof(IDictionary).GetTypeInfo()` be cached into a static field? Or is the reflection in it cheap enough to not worry about it?